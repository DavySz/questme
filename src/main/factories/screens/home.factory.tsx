import { HomeContainer } from "@presentation/screens";
import { ReactNode } from "react";

export function MakeHomeScreen(): ReactNode {
  return <HomeContainer />;
}
