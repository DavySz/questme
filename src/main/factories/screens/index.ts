export * from "./onboarding-factory";
export * from "./splash-factory";
export * from "./initial-factory";
export * from "./login-factory";
export * from "./sign-up-factory";
