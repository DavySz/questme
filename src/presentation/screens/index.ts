export * from "./home/home.container";
export * from "./splash/splash.container";
export * from "./onboarding/onboarding.container";
export * from "./initial/initial.container";
export * from "./login/login.container";
export * from "./sign-up/sign-up.container";
export * from "./reset-password/reset-password.container";
export * from "./new-password/new-password.container";
export * from "./sign-up-email";
