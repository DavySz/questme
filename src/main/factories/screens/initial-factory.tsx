import { InitialContainer } from "@presentation/screens";
import { ReactNode } from "react";

export function MakeInitialScreen(): ReactNode {
  return <InitialContainer />;
}
