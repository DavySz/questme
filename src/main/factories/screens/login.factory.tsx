import { LoginContainer } from "@presentation/screens";
import { ReactNode } from "react";

export function MakeLoginScreen(): ReactNode {
  return <LoginContainer />;
}
