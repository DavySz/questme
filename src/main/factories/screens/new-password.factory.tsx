import { NewPasswordContainer } from "@presentation/screens";
import { ReactNode } from "react";

export function MakeNewPasswordScreen(): ReactNode {
  return <NewPasswordContainer />;
}
