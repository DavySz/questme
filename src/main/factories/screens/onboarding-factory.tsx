import { OnboardingContainer } from "@presentation/screens";
import { ReactNode } from "react";

export function MakeOnboardingScreen(): ReactNode {
  return <OnboardingContainer />;
}
