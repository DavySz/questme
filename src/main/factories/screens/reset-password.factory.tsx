import { ResetPasswordContainer } from "@presentation/screens";
import { ReactNode } from "react";

export function MakeResetPasswordScreen(): ReactNode {
  return <ResetPasswordContainer />;
}
