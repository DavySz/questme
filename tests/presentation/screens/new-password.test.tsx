import { fireEvent, render } from "@testing-library/react-native";
import { TestingPublicNavigator, customRender } from "../../utils";
import { NewPasswordContainer } from "@presentation/screens";

describe("NewPassword", () => {
  test("Should render screen without errors", () => {
    const tree = customRender(<NewPasswordContainer />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("Should navigate to correct screen when button reset password is pressed", async () => {
    const { getByTestId, findByTestId } = render(
      <TestingPublicNavigator currentRoute="new-password" />
    );

    const buttonResetPassword = getByTestId("button-reset-password");
    fireEvent.press(buttonResetPassword);

    const screen = await findByTestId("initial-screen-content");

    expect(screen).toBeTruthy();
  });
});
