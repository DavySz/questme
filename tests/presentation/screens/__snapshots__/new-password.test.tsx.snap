// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewPassword Should render screen without errors 1`] = `
<View
  style={
    {
      "backgroundColor": "#EFEEFC",
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
    }
  }
  testID="default-container"
  variant="secondary"
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
        "width": "100%",
      }
    }
  >
    <Text
      color="neutral-black"
      style={
        {
          "color": "#0C092A",
          "fontFamily": "Rubik_500Medium",
          "fontSize": 24,
          "lineHeight": 36,
        }
      }
      testID="header-title"
      variant="heading-3"
    >
      New Password
    </Text>
  </View>
  <RCTScrollView
    contentContainerStyle={
      {
        "alignItems": "center",
        "flex": 1,
        "gap": 24,
        "padding": 24,
      }
    }
    showsVerticalScrollIndicator={false}
    style={{}}
    variant="secondary"
  >
    <View>
      <Text
        color="neutral-gey-2"
        style={
          {
            "color": "#858494",
            "fontFamily": "Rubik_400Regular",
            "fontSize": 16,
            "lineHeight": 24,
          }
        }
        variant="body-normal-regular"
      >
        Your new password must be different from previous used passwords.
      </Text>
      <View
        style={
          {
            "gap": 16,
            "paddingBottom": 0,
            "paddingLeft": 24,
            "paddingRight": 24,
            "paddingTop": 0,
          }
        }
      >
        <View
          style={{}}
        >
          <View
            style={
              {
                "marginBottom": 8,
              }
            }
          >
            <Text
              color="neutral-black"
              style={
                {
                  "color": "#0C092A",
                  "fontFamily": "Rubik_400Regular",
                  "fontSize": 14,
                  "lineHeight": 20,
                }
              }
              variant="body-small-regular"
            >
              Password
            </Text>
          </View>
          <View
            isFocused={false}
            style={
              {
                "backgroundColor": "#FFF",
                "borderBottomLeftRadius": 20,
                "borderBottomRightRadius": 20,
                "borderColor": "#EFEEFC",
                "borderStyle": "solid",
                "borderTopLeftRadius": 20,
                "borderTopRightRadius": 20,
                "borderWidth": 2,
                "flexDirection": "row",
                "gap": 16,
                "paddingBottom": 16,
                "paddingLeft": 16,
                "paddingRight": 16,
                "paddingTop": 16,
                "width": "100%",
              }
            }
            testID="input-wrapper"
          >
            <SvgMock />
            <TextInput
              onBlur={[Function]}
              onFocus={[Function]}
              placeholder="Your password"
              style={
                {
                  "fontFamily": "Rubik_400Regular",
                  "fontSize": 16,
                  "width": "100%",
                }
              }
              testID="custom-input"
            />
          </View>
        </View>
        <View
          style={{}}
        >
          <View
            style={
              {
                "marginBottom": 8,
              }
            }
          >
            <Text
              color="neutral-black"
              style={
                {
                  "color": "#0C092A",
                  "fontFamily": "Rubik_400Regular",
                  "fontSize": 14,
                  "lineHeight": 20,
                }
              }
              variant="body-small-regular"
            >
              Confirm Password
            </Text>
          </View>
          <View
            isFocused={false}
            style={
              {
                "backgroundColor": "#FFF",
                "borderBottomLeftRadius": 20,
                "borderBottomRightRadius": 20,
                "borderColor": "#EFEEFC",
                "borderStyle": "solid",
                "borderTopLeftRadius": 20,
                "borderTopRightRadius": 20,
                "borderWidth": 2,
                "flexDirection": "row",
                "gap": 16,
                "paddingBottom": 16,
                "paddingLeft": 16,
                "paddingRight": 16,
                "paddingTop": 16,
                "width": "100%",
              }
            }
            testID="input-wrapper"
          >
            <SvgMock />
            <TextInput
              onBlur={[Function]}
              onFocus={[Function]}
              placeholder="Confirm your password"
              style={
                {
                  "fontFamily": "Rubik_400Regular",
                  "fontSize": 16,
                  "width": "100%",
                }
              }
              testID="custom-input"
            />
          </View>
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#6A5AE0",
            "borderBottomLeftRadius": 20,
            "borderBottomRightRadius": 20,
            "borderTopLeftRadius": 20,
            "borderTopRightRadius": 20,
            "flexDirection": "row",
            "justifyContent": "center",
            "opacity": 1,
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
            "width": "100%",
          }
        }
        testID="button-reset-password"
      >
        <Text
          color="neutral-white"
          style={
            {
              "color": "#FFF",
              "fontFamily": "Rubik_500Medium",
              "fontSize": 16,
              "lineHeight": 24,
            }
          }
          variant="body-normal-medium"
        >
          Reset Password
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
